django_mongodb/compiler.py

```diff
@@ -435,11 +435,19 @@ def project_field(column):
 
     @cached_property
     def collection_name(self):
-        base_table = next(
-            v
-            for k, v in self.query.alias_map.items()
-            if isinstance(v, BaseTable) and self.query.alias_refcount[k]
-        )
+        try:
+            base_table = next(
+                v
+                for k, v in self.query.alias_map.items()
+                if isinstance(v, BaseTable) and self.query.alias_refcount[k]
+            )
+        except StopIteration:
+            # Use a dummy collection if the query doesn't specify a table
+            # (such as Constraint.validate() with a condition).
+            query = self.query_class(self)
+            query.aggregation_pipeline = [{"$facet": {"__null": []}}]
+            self.subqueries.insert(0, query)
+            return "__null"
```

Comment:
Well, the facet execution will have a very short pipeline, is just a way to insert a document so the subqueries won't get empty.
I mean
just this example:
```SQL
select 1
where exists(some subquery checking that an index exists)
```

This is translated as:
```JSON
[{"$facet": {"__null": []}},
 {"$lookup": {"the subquery"}}
]
```

If a query over an empty collection is executed, the result will be 0 rows. To avoid that, I put a single tiny document